dronekit/__init__.py

```diff
@@ -1162,7 +1162,8 @@ def listener(self, name, m):
             self._armed = (m.base_mode & mavutil.mavlink.MAV_MODE_FLAG_SAFETY_ARMED) != 0
             self.notify_attribute_listeners('armed', self.armed, cache=True)
             if self._master.mode_mapping() != None:
-                self._flightmode = {v: k for k, v in self._master.mode_mapping().items()}[m.custom_mode]
+                if m.custom_mode in {v: k for k, v in self._master.mode_mapping().items()}:
```

Comment:
would prefer to avoid evaluating `{v: k for k, v in self._master.mode_mapping().items()}` twice